### DISCUSSION TITLE
Binary type from point + size

### DISCUSSION THREAD

**Discussion by fleon-psi** [nlohmann/json#2379] [Q&A] [answered]
I have arrays of numbers (mix of std::vector<float> and std::vector<uint32_t>). I would like to place them into a single CBOR file.

I'm able to do it by making a new std::vector<uint8_t> and copying contest of float/uint array there and then using nlpohmann::json::binary(), but this adds extra memory copy. 

Is there a simpler way to do it, e.g. by providing simple pointer and number of bytes?

**Comment by nlohmann** [comment, marked as answer]
No, this is not possible: all values are stored inside the `json` class - it is not possible to store references.

**Reply by fleon-psi** [reply]
Thank you!
